src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -275,6 +276,23 @@ public string EndWriteTagHelperAttribute()
             return content;
         }
 
+        public virtual void PushWriter(TextWriter writer)
```

Comment:
I'm actually not 100% certain about this. This works today because we codegen inline lambdas for things like templates. If ever wanted to turn something like that into a library, the visibility check would block us.